### PR TITLE
[Release] Convert Horovod and SGD release tests

### PR DESCRIPTION
**Pull request by amogkam**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/BUILD [hunk 1]
```diff
@@ -800,6 +800,20 @@ py_test(
     args = ["--smoke-test"]
 )
 
+# --------------------------------------------------------------------
+# Tune related tests from the ../../../../release directory.
+# Please keep these sorted alphabetically.
+# --------------------------------------------------------------------
+
+py_test(
+    name = "horovod_test",
```
Comment (amogkam) [inline python/ray/tune/BUILD:809]:
```suggestion
    name = "horovod_cifar_pbt_example",
```

**Comment by amogkam** [comment]
@richardliaw do you know how to get the symlink working for the horovod smoke test?

**Comment by richardliaw** [comment]
@amogkam I took a pass, should work now hopefully

**Comment by richardliaw** [comment]
ping when tests pass?

**Review by krfricke — APPROVED** [review]
Looks good to me. Have you run this via `e2e.py` locally?

**Comment by richardliaw** [comment]
Hmm @amogkam @krfricke  I wasn't quite sure how to resolve these conflicts given the #15913 just merged and seemed to have a bit of overlap.

**Comment by krfricke** [comment]
Hey @amogkam I resolved the conflicts and I think this should be good to go. Can you take one more look?

**Comment by amogkam** [comment]
Thanks a lot @krfricke! I'll look into the failing test. But we shouldn't merge this in yet until Horovod can be built on App configs. Once we do that I'll confirm that the release test works with e2e.py locally and then we can merge this.

**Comment by krfricke** [comment]
Test seems to pass fine on buildkite: https://buildkite.com/ray-project/periodic-ci/builds/198#_

Feel free to merge after fixing the unit test

**Comment by richardliaw** [comment]
Is this working?

**Comment by amogkam** [comment]
I think we need to merge this first https://github.com/ray-project/ray/pull/16581/files, and then pull master again

**Pull request merged**